plugins/extractors/googleworkspace/googleworkspace.go

```diff
@@ -0,0 +1,219 @@
+	Tags:         []string{"platform", "extractor"},
+	Summary:      summary,
+}
+
+// Extractor manages the extraction of data from the extractor
+type Extractor struct {
+	plugins.BaseExtractor
+	logger      log.Logger
+	config      Config
+	TokenSource oauth2.TokenSource
+	Client      *admin.Service
+	emit        plugins.Emit
+}
+
+// New returns a pointer to an initialized Extractor Object
+func New(logger log.Logger) *Extractor {
+	e := &Extractor{
+		logger: logger,
+	}
+	e.BaseExtractor = plugins.NewBaseExtractor(info, &e.config)
+
+	return e
+}
+
+// Init initializes the extractor
+func (e *Extractor) Init(ctx context.Context, config plugins.Config) (err error) {
+	if err = e.BaseExtractor.Init(ctx, config); err != nil {
+		return err
+	}
+
+	jwtConfig, err := google.JWTConfigFromJSON([]byte(e.config.ServiceAccountJSON), admin.AdminDirectoryUserScope)
+	if err != nil {
+		return plugins.InvalidConfigError{}
+	}
+
+	if e.config.UserEmail == "" {
+		return plugins.InvalidConfigError{}
+	}
+	jwtConfig.Subject = e.config.UserEmail
+	ts := jwtConfig.TokenSource(ctx)
+
+	e.TokenSource = ts
+
+	return
+}
+
+// Extract extracts the data from the extractor
+// The data is returned as a list of assets.Asset
+func (e *Extractor) Extract(ctx context.Context, emit plugins.Emit) (err error) {
+	var status string
+	e.emit = emit
+	r, err := FetchUsers(ctx, e.TokenSource)
+	if err != nil {
+		return err
+	}
+
+	if len(r.Users) == 0 {
+		e.logger.Info("No users found.\n")
+		return nil
+	}
+
+	for _, u := range r.Users {
+		if !u.Suspended {
+			status = "not suspended"
+		} else {
+			status = "suspended"
+		}
+
+		var userAttributes = make(map[string]interface{})
+		userAttributes = buildOrgAttributes(userAttributes, u.Organizations)
+		userAttributes = buildRelationsAttributes(userAttributes, u.Relations)
+		userAttributes = buildCustomSchemasAttributes(userAttributes, u.CustomSchemas)
+		userAttributes["org_unit_path"] = u.OrgUnitPath
+		userAttributes["aliases"] = strings.Join(u.Aliases, ",")
+
+		e.emit(models.NewRecord(&assetsv1beta1.User{
+			Resource: &commonv1beta1.Resource{
+				Service: "google workspace",
+				Name:    u.Name.FullName,
+				Urn:     u.PrimaryEmail,
+			},
+			Email:    u.PrimaryEmail,
+			FullName: u.Name.FullName,
+			Status:   status,
+			Properties: &facetsv1beta1.Properties{
+				Attributes: utils.TryParseMapToProto(userAttributes),
+			},
+		}))
+	}
+
+	return nil
+}
+
+// init registers the extractor to catalog
+func init() {
+	if err := registry.Extractors.Register("googleworkspace", func() plugins.Extractor {
+		return New(plugins.GetLog())
+	}); err != nil {
+		panic(err)
+	}
+}
+
+func FetchUsers(ctx context.Context, ts oauth2.TokenSource) (*admin.Users, error) {
+	srv, err := admin.NewService(ctx, option.WithTokenSource(ts))
+	if err != nil {
+		return nil, err
+	}
+
+	r, err := srv.Users.List().Customer("my_customer").Do()
+	if err != nil {
+		return nil, errors.Wrap(err, "Unable to retrieve users in domain")
```

Comment:
following golang code review guide, we should not start error string with capitalized letters. [[ref](https://github.com/golang/go/wiki/CodeReviewComments#error-strings)]